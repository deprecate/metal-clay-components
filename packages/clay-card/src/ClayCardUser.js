import './ClayCardBase';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayCardUser.soy.js';

/**
 * Metal ClayCardUser component.
 */
class ClayCardUser extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayCardUser.STATE = {
	/**
	 * Actions in the dropdown on card.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?array|undefined}
	 * @default undefined
	 */
	actionsItems: Config.arrayOf(
		Config.shapeOf({
			label: Config.string().required(),
			separator: Config.bool().value(false),
			url: Config.string(),
		})
	),

	/**
	 * Transform card with checkbox.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?array|undefined}
	 * @default undefined
	 */
	checkbox: Config.shapeOf({
		checked: Config.bool().value(false),
		disabled: Config.bool().value(false),
		indeterminate: Config.bool().value(false),
		name: Config.string(),
		value: Config.string(),
	}),

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 *
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	icon: Config.string(),

	/**
	 * Alt the image.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	imageAlt: Config.string(),

	/**
	 * The src to be applied to the background or user card image.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	imageSrc: Config.string(),

	/**
	 * Render label in the ClayLabel element.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	label: Config.string(),

	/**
	 * Style of the ClayLabel.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string}
	 * @default secondary
	 */
	labelStyle: Config.oneOf([
		'danger',
		'info',
		'secondary',
		'success',
		'warning',
	]).value('secondary'),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {!string}
	 * @default undefined
	 */
	spritemap: Config.string().required(),

	/**
	 * Subtitle of the card.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	subtitle: Config.string(),

	/**
	 * Title of the card.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Render label in the ClaySticker element.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string|undefined}
	 * @default undefined
	 */
	type: Config.string(),

	/**
	 * Style of the ClaySticker element.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string}
	 * @default primary
	 */
	typeStyle: Config.oneOf([
		'danger',
		'dark',
		'info',
		'light',
		'primary',
		'secondary',
		'success',
		'warning',
	]).value('primary'),

	/**
	 * Transform title in the url.
	 * @instance
	 * @memberof ClayCardUser
	 * @type {?string}
	 * @default undefined
	 */
	url: Config.string(),
};

defineWebComponent('clay-card-user', ClayCardUser);

Soy.register(ClayCardUser, templates);

export {ClayCardUser};
export default ClayCardUser;
