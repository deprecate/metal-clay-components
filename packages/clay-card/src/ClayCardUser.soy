{namespace ClayCardUser}

{template .render}
    {@param? actionItems: list<[
        label: string,
        separator: bool,
        url: string
    ]>}
    {@param? disabled: bool}
    {@param? elementClasses: string}
    {@param? id: string}
    {@param? imageAlt: string}
		{@param? imageSrc: string}
    {@param? inputName: string}
    {@param? inputValue: string}
    {@param? selectable: bool}
    {@param? selected: bool}
    {@param? spritemap: string}
    {@param? subtitle: string}
    {@param? title: string}
    {@param? type: string}
		{@param? typeStyle: string}
    {@param? url: string}


    {call ClayCardBase.render}
        {param actionItems: $actionItems /}
        {param disabled: $disabled /}
        {param elementClasses: $elementClasses /}
        {param id: $id /}
        {param imageAlt: $imageAlt /}
        {param imageSrc: $imageSrc /}
        {param inputName: $inputName /}
        {param inputValue: $inputValue /}
        {param ref: 'cardUser' /}
        {param selectable: $selectable /}
        {param selected: $selected /}
        {param spritemap: $spritemap /}
        {param subtitle: $subtitle /}
        {param title: $title /}
        {param type: $type /}
        {param typeStyle: $typeStyle /}
        {param url: $url /}
        {param variant: 'user' /}
    {/call}
{/template}

/**
 * This renders the information area of the card.
 */
{deltemplate ClayCardBase.DescriptiveArea variant="'user'"}
	{@param? actionItems: list<[
		label: string,
		separator: bool,
		url: string
	]>}
	{@param? label: string}
	{@param? labelStyle: string}
	{@param? spritemap: string}
	{@param? subtitle: string}
	{@param? title: string}
	{@param? url: string}

	<div class="card-body">
		<div class="card-row">
			<div class="flex-col flex-col-expand">
				{if $url}
					{call ClayLink.render}
						{param elementClasses: 'card-title text-truncate' /}
						{param href: $url /}
						{param label: $title /}
					{/call}
				{elseif $title}
					<div class="card-title text-truncate" title="{$title}">{$title}</div>
				{/if}

				{if $subtitle}
					<div class="card-subtitle text-truncate" title="{$subtitle}">{$subtitle}</div>
				{/if}
			</div>

			{if $actionItems and $spritemap}
				<div class="flex-col">
					{call ClayActionsDropdown.render}
						{param items: $actionItems /}
						{param spritemap: $spritemap /}
					{/call}
				</div>
			{/if}
		</div>
	</div>
{/deltemplate}

/**
 * Render the visual area of the card.
 */
{deltemplate ClayCardBase.VisualArea variant="'user'"}
	{@param? icon: string}
	{@param? imageAlt: string}
	{@param? imageSrc: string}
	{@param? spritemap: string}
	{@param? type: string}
	{@param? typeStyle: string}
	{@param? variant: string}

	{let $stickerStyle: $typeStyle ?: 'primary' /}

	{let $classesAspectRatio kind="text"}
		aspect-ratio-item-center-middle aspect-ratio-item-fluid
	{/let}

	<div class="aspect-ratio card-item-first">
		<div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
			{call ClaySticker.render}
				{param label kind="html"}
					{if $imageSrc}
						{let $imageAttributes kind="attributes"}
							alt="
								{if $imageAlt}
									{$imageAlt}
								{/if}
							"

							class="img-fluid"
							src="{$imageSrc}"
						{/let}

						<img {$imageAttributes} />
					{elseif $type}
						{$type}
					{/if}
				{/param}

				{param shape: 'circle' /}
				{param size: 'xl' /}
				{param style: $stickerStyle /}
			{/call}
		</div>
	</div>
{/deltemplate}