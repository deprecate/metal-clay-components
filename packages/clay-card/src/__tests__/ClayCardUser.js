import ClayCardUser from '../ClayCardUser';

let component;
let spritemap = 'icons.svg';
let actionsItems = [
	{
		label: 'Edit',
		url: '#1',
	},
	{
		label: 'Save',
		url: '#1',
	},
];

describe('ClayCardUser', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayCardUser({
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with classes', () => {
		component = new ClayCardUser({
			spritemap: spritemap,
			elementClasses: 'my-custom-class',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with id', () => {
		component = new ClayCardUser({
			spritemap: spritemap,
			id: 'myId',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with actionsItems', () => {
		component = new ClayCardUser({
			actionsItems: actionsItems,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with checkbox checked', () => {
		component = new ClayCardUser({
			checkbox: {
				checked: true,
			},
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with checkbox disabled', () => {
		component = new ClayCardUser({
			checkbox: {
				checked: true,
				disabled: true,
			},
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with checkbox indeterminate', () => {
		component = new ClayCardUser({
			checkbox: {
				indeterminate: true,
			},
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with checkbox `name`', () => {
		component = new ClayCardUser({
			checkbox: {
				checked: true,
				name: 'checkbox01',
			},
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with checkbox `value`', () => {
		component = new ClayCardUser({
			checkbox: {
				checked: true,
				value: 'checkbox',
			},
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with icon', () => {
		component = new ClayCardUser({
			icon: 'documents',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with imageAlt', () => {
		component = new ClayCardUser({
			imageAlt: 'thumbnail',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with imageSrc', () => {
		component = new ClayCardUser({
			imageSrc: 'thumbnail_coffee.jpg',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with label', () => {
		component = new ClayCardUser({
			label: 'Approved',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with labelStyle `danger color`', () => {
		component = new ClayCardUser({
			label: 'Approved',
			labelStyle: 'danger',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with labelStyle `info color`', () => {
		component = new ClayCardUser({
			label: 'Approved',
			labelStyle: 'info',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with labelStyle `secondary color`', () => {
		component = new ClayCardUser({
			label: 'Approved',
			labelStyle: 'secondary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with labelStyle `success color`', () => {
		component = new ClayCardUser({
			label: 'Approved',
			labelStyle: 'success',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with labelStyle `warning color`', () => {
		component = new ClayCardUser({
			label: 'Approved',
			labelStyle: 'warning',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with subtitle', () => {
		component = new ClayCardUser({
			subtitle: 'Author Action',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with title', () => {
		component = new ClayCardUser({
			subtitle: 'empty-background.png',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type', () => {
		component = new ClayCardUser({
			type: 'DOC',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `danger color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'danger',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `dark color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'dark',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `info color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'info',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `light color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'light',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `primary color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'primary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `secondary color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'secondary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `success color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'success',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with type `warning color`', () => {
		component = new ClayCardUser({
			type: 'DOC',
			typeStyle: 'warning',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardUser with url', () => {
		component = new ClayCardUser({
			url: '#1',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});
});
