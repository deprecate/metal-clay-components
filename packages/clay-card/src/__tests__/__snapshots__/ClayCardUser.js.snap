// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayCardUser should render a ClayCardUser with actionsItems 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
      <div class="flex-col">
        <div class="dropdown dropdown-action">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
              <use xlink:href="icons.svg#ellipsis-v"></use>
            </svg>
          </button>
          <div class="dropdown-menu">
            <ul class="list-unstyled">
              <li>
                <a class=" dropdown-item" href="#1">Edit</a>
              </li>
              <li>
                <a class=" dropdown-item" href="#1">Save</a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with checkbox \`name\` 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" name="checkbox01" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with checkbox \`value\` 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" value="checkbox" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with checkbox checked 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with checkbox disabled 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" disabled="disabled" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with checkbox indeterminate 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first">
          <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
            <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
          </div>
        </div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with classes 1`] = `
<div class="card card-type-asset my-custom-class">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with icon 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents">
        <use xlink:href="icons.svg#documents"></use>
      </svg>
    </div>
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with id 1`] = `
<div class="card card-type-asset" id="myId">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with imageAlt 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with imageSrc 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl">
        <img class="img-fluid" src="thumbnail_coffee.jpg">
      </span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with label 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with labelStyle \`danger color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-danger">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with labelStyle \`info color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-info">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with labelStyle \`secondary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with labelStyle \`success color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-success">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with labelStyle \`warning color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-warning">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with subtitle 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-subtitle text-truncate" title="Author Action">Author Action</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with title 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-title text-truncate" title="empty-background.png">empty-background.png</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`danger color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-danger rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`dark color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-dark rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`info color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-info rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`light color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-light rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`primary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`secondary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-secondary rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`success color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-success rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type \`warning color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-warning rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with type 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl">DOC</span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render a ClayCardUser with url 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <a class=" card-title text-truncate" href="#1"></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardUser should render the default markup 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <span class="sticker sticker-primary rounded-circle sticker-xl"></span>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;
